### DISCUSSION TITLE
cant disable response cache when using browser back button

### DISCUSSION THREAD

**Discussion by maxodoble** [unpoly/unpoly#644] [General]
hello,

i am trying to disable the response cache when using the browser back button, and am getting a potentially stale state because of this.

the log shows this line when using the back button:

`up:location:restore - Restoring location /dokumente/3e351d1....`

`up.request() - Re-using previous request to GET /dokumente/...`

how can i disable this cache when using the back button?  

what i tried is setting these config settings

`up.network.config.autoCache = () => false`
and/or
`up.network.config.cacheExpireAge = 1_000`

but this only seems to work when clicking on the element, and not when navigating with browser back/forward: 

- clicking on link: i can  see the xhr get request to the server
- using back or forward browser button: no xhr request, uses the cache (with `...Re-using previous request...` line visible in log)

the html code element  i am using (golang templ serverside):

`<div class="flex-1" up-target="#main" up-href={ "/dokumente/" + item.Uid }>...child elems...</div>`

any ideas?

**Comment by maxodoble** [comment]
found one way that works for my usecase:

`up.network.config.cacheSize=1`

with this setting the cache is deactivated for manual navigation (clicking on a link or element) or when navigating with browser back/forward.

 if there is any way to use a less "brute force" way to reliably deactivate response caches, please let me know :wink:

**Comment by triskweline** [comment]
Hi @maxodoble. I checked the code and it looks like there is no great way to do this currently. I put it on the backlog.

**Reply by maxodoble** [reply]
Thanks for looking into this, and much more thanks for unpoly: What a gem of a tool for web development!

Max

**Reply by triskweline** [reply]
Thank you @maxodoble :heart:

**Comment by triskweline** [comment]
The following changes are now available in [Unpoly 3.8.0](https://github.com/unpoly/unpoly/discussions/644):

- When the user presses the back button and Unpoly restores content from the cache, that content is then [revalidated](https://unpoly.com/caching#revalidation) with the server. This ensures that content is shown with the most recent data.
- Listeners to [`up:location:restore`](https://unpoly.com/up:location:restore) may now mutate the `event.renderOptions` event to customize the render pass that is about to restore content. With this you can, for example, disable caching by setting `event.renderOptions.cache = false`.

**Comment by hpvd** [comment]
just gave browser back button a try on unpoly.com
-> there was an error:
https://github.com/unpoly/unpoly/issues/647
